test/integration/app/libraries/iis_app.rb

```diff
@@ -0,0 +1,124 @@
+# encoding: utf-8
+# frozen_string_literal: true
+# check for site in IIS
+# Usage:
+# describe iis_app('/myapp', 'Website') do
+#   it { should exist }
+#   it { should have_application_pool('MyAppPool') }
+#   it { should have_protocols('http') }
+#   it { should have_site_name('Default Web Site') }
+#   it { should have_physical_path('C:\\inetpub\\wwwroot\\myapp') }
+#   it { should have_path('\\My Application') }
+# end
+#
+# Note: this is only supported in windows 2012 and later
+
+class IisApp < Inspec.resource(1)
+  name 'iis_app'
+  desc 'Tests IIS application configuration on windows. Supported in server 2012+ only'
+  example "
+    describe iis_app('/myapp', 'Default Web Site') do
+      it { should exist }
+      it { should have_application_pool('MyAppPool') }
+      it { should have_protocols('http') }
+      it { should have_site_name('Default Web Site') }
+      it { should have_physical_path('C:\\inetpub\\wwwroot\\myapp') }
+      it { should have_path('\\My Application') }
+    end
+  "
+
+  def initialize(path, site_name)
+    @path = path
+    @site_name = site_name
+    @cache = nil
+
+    @app_provider = AppProvider.new(inspec)
+
+    # verify that this resource is only supported on Windows
+    return skip_resource 'The `iis_app` resource is not supported on your OS.' if inspec.os[:family] != 'windows'
```

Comment:
We have some helper methods that make OS filtering easier than expecting people to know all the families, etc., and we're encouraging community members submitting resources for core InSpec to use them.

In this case, you should be able to do `unless inspec.os.windows?`